test/object-store/sync/sync_test_utils.cpp

```diff
@@ -377,20 +377,17 @@ struct TestServerClientReset : public TestClientReset {
             auto realm2 = Realm::get_shared_realm(m_remote_config);
             auto session2 = sync_manager->get_existing_session(realm2->config().path);
 
-            for (int i = 0; i < 2; ++i) {
-                wait_for_download(*realm2);
-                realm2->begin_transaction();
-                auto table = get_table(*realm2, "object");
-                auto col = table->get_column_key("value");
-                table->begin()->set(col, i + 5);
-                if (i == 1 && m_make_remote_changes) {
-                    m_make_remote_changes(realm2);
-                }
-                realm2->commit_transaction();
-                wait_for_upload(*realm2);
-                server.advance_clock(10s);
+            wait_for_download(*realm2);
+            realm2->begin_transaction();
+            auto table = get_table(*realm2, "object");
+            auto col = table->get_column_key("value");
+            table->begin()->set(col, 6);
+            if (m_make_remote_changes) {
+                m_make_remote_changes(realm2);
             }
-            server.advance_clock(10s);
+            realm2->commit_transaction();
```

Comment:
Ok, based on your explanation, makes sense to commit. Although, the fact that a different version of the server might change the behaviour of the tests is a little daunting.

Reply:
There are three distinct versions of the client reset test harness. Here is some background to why we have them:
1) `TestServerClientReset`: We want to remove this entirely eventually because the test server here is an out of date and unmaintained version of sync. This is used when running sync tests without baas available such as Jenkins MacOS nodes (ie where the race was discovered).
2) `BaasClientReset`. Runs a full integration test against an actual MongoDB server instance. This is slow because we have to wait for sync and proper app setup and teardown. It is used in the evergreen suite, and for one Jenkins test where we spin up the baas in docker.
3) `FakeLocalClientReset`: simulates the operations locally without any sync server required. This is not a full integration test, but it is the fastest because it does not have to wait for sync at all.

Reply:
OK, this makes sense. I wonder if we don't have option 1 servers that are actually functioning in production, why keep the test then? Wouldn't it make much more sense to have option 3 as Unit Test and option 2 as Integration?... anyway I guess this is beyond the scope of this PR :-)